suave/gen/main.go

```diff
@@ -243,6 +265,13 @@ type {{.Name}} struct {
 	{{range .Fields}}{{title .Name}} {{typ3 .Typ}}
 	{{end}}
 }
+
+type {{.Name | lower}}Marshaling struct {
```

Comment:
Do you need to do this as part of the code generation?

Could you create another file `suave_structs_encoding.go` which is not code generated that does the encoding overriding?